AutomatedTesting/Gem/PythonTests/scripting/EditMenu_Default_UndoRedo.py

```diff
@@ -49,58 +53,55 @@ class EditMenu_Default_UndoRedo:
-
-        # 5) Trigger Undo action and verify if variable is removed in Variable Manager
-        undo_redo_action = self.sc_editor.findChild(QtWidgets.QAction, "action_Undo")
-        undo_redo_action.trigger()
-        row_count = 0
-        Report.result(Tests.undo_worked, helper.wait_for_condition(
-            lambda: self.get_graph_variables_row_count(graph_variables) == row_count, WAIT_TIME_3))
-
-        # 6) Trigger Redo action and verify if variable is re-added in Variable Manager
-        undo_redo_action = self.sc_editor.findChild(QtWidgets.QAction, "action_Redo")
-        undo_redo_action.trigger()
-        row_count = 1
-        Report.result(Tests.redo_worked, helper.wait_for_condition(
-            lambda: self.get_graph_variables_row_count(graph_variables) == row_count, WAIT_TIME_3))
-
-        # 7) Close SC window
-        general.close_pane(SCRIPT_CANVAS_UI)
 
 
-test = EditMenu_Default_UndoRedo()
-test.run_test()
+    # Preconditions
+    from PySide2 import QtWidgets
+    import PySide2.QtCore as QtCore
+    from editor_python_test_tools.utils import TestHelper as helper
+    from editor_python_test_tools.utils import Report
+    import azlmbr.legacy.general as general
+    import pyside_utils
+    import editor_python_test_tools.EditorQtContainer as qtContainer
+    import editor_python_test_tools.scripting_tools_qt as scripting_tools_qt
+    from scripting_utils.scripting_constants import (WAIT_TIME_3, SCRIPT_CANVAS_UI, VARIABLE_TYPES,
+                                                     GRAPH_VARIABLES_QT)
+
+    general.idle_enable(True)
+
+    # 1) Open Script Canvas window and initialize the qt SC objects
+    general.open_pane(SCRIPT_CANVAS_UI)
+    helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI), WAIT_TIME_3)
+    qtObjects = scripting_tools_qt.initialize_script_qt_canvas_objects()
+
+    # 3) Create Graph
+    scripting_tools_qt.create_new_sc_graph(qtObjects.sc_editor_main_window)
+
+    # 4) Create and verify the new variable exists in variable manager
+    scripting_tools_qt.create_new_variable(qtObjects, VARIABLE_TYPES[0])
+    graph_variables = qtObjects.variable_manager.findChild(QtWidgets.QTableView, GRAPH_VARIABLES_QT)
+    row_count = 1
+    Report.result(Tests.variable_created, helper.wait_for_condition(
+        lambda: get_graph_variables_row_count(graph_variables, QtCore) == row_count, WAIT_TIME_3))
+
+    # 5) Trigger Undo action and verify if variable is removed in Variable Manager
+    undo_redo_action = qtObjects.sc_editor.findChild(QtWidgets.QAction, "action_Undo")
+    undo_redo_action.trigger()
+    row_count = 0
+    Report.result(Tests.undo_worked, helper.wait_for_condition(
+        lambda: get_graph_variables_row_count(graph_variables, QtCore) == row_count, WAIT_TIME_3))
+
+    # 6) Trigger Redo action and verify if variable is re-added in Variable Manager
+    undo_redo_action = qtObjects.sc_editor.findChild(QtWidgets.QAction, "action_Redo")
```

Comment:
Let's create a helper function here to wrap this behavior in the scripting_tools_qt library.

Reply:
put this into reusable tools function